iothub/device/src/AuthenticationWithTokenRefresh.cs

```diff
@@ -18,8 +18,8 @@ public abstract class AuthenticationWithTokenRefresh : IAuthenticationMethod
         private readonly int _timeBufferPercentage;
 
         private int _bufferSeconds;
-        private SemaphoreSlim _lock = new SemaphoreSlim(1);
-        private string _token = null;
+        private readonly SemaphoreSlim _lock = new SemaphoreSlim(1);
```

Comment:
Is this disposed with the class/client? I ask because often we dispose them and set them to null but if this is readonly then we wouldn't be setting it to null.

Reply:
That is an excellent point, and by that I mean this class should be disposable since it has disposable fields. Changing.